src/stub/core_impl/register_info.rs

```diff
@@ -0,0 +1,105 @@
+                            Some(Register {
+                                name,
+                                alt_name,
+                                bitsize,
+                                offset,
+                                encoding,
+                                format,
+                                set,
+                                gcc,
+                                dwarf,
+                                generic,
+                                container_regs,
+                                invalidate_regs,
+                            }) => {
+                                // TODO: replace this with a try block (once stabilized)
+                                (|| {
+                                    res.write_str("name:")?;
+                                    res.write_str(name)?;
+                                    res.write_str(";alt-name:")?;
+                                    res.write_str(alt_name)?;
+                                    res.write_str(";bitsize:")?;
+                                    res.write_dec(bitsize)?;
+                                    res.write_str(";offset:")?;
+                                    res.write_dec(offset)?;
+                                    res.write_str(";encoding:")?;
+                                    res.write_str(encoding)?;
+                                    res.write_str(";format:")?;
+                                    res.write_str(format)?;
+                                    res.write_str(";set:")?;
+                                    res.write_str(set)?;
+                                    if let Some(gcc) = gcc {
+                                        res.write_str(";gcc:")?;
+                                        res.write_dec(gcc)?;
+                                    }
+                                    if let Some(dwarf) = dwarf {
+                                        res.write_str(";dwarf:")?;
+                                        res.write_dec(dwarf)?;
+                                    }
+                                    if let Some(generic) = generic {
+                                        res.write_str(";generic:")?;
+                                        res.write_str(generic)?;
+                                    }
+                                    if let Some(c_regs) = container_regs {
+                                        write_register!(c_regs, ";container-regs:");
+                                    }
+                                    if let Some(i_regs) = invalidate_regs {
+                                        write_register!(i_regs, ";invalidate-regs:");
+                                    }
+                                    res.write_str(";")
+                                })()
+                            }
+                            // This will end the 'qRegisterInfo' query
+                            None => res.write_str("E45"),
```

Comment:
The `qRegisterInfo` request loop starts [here](https://github.com/llvm/llvm-project/blob/369ce54bb302f209239b8ebc77ad824add9df089/lldb/source/Plugins/Process/gdb-remote/ProcessGDBRemote.cpp#L444) and goes on till
`response_type != StringExtractorGDBRemote::eResponse`. It will get reassigned in the loop by `response.GetResponseType();` which is defined [here](https://github.com/llvm/llvm-project/blob/369ce54bb302f209239b8ebc77ad824add9df089/lldb/source/Utility/StringExtractorGDBRemote.cpp#L18) and returns `eError` if we parse an `E` and the packet has size 3
```
case 'E':
    if (isxdigit(m_packet[1]) && isxdigit(m_packet[2])) {
      if (m_packet.size() == 3)
        return eError;
```
So we could indeed return something else here (maybe a unique value to gdbstub?) :laughing:

Reply:
Haha, nice. Honestly, I'd just add a little inline comment summarising what we've discovered here, and leave the value as is. After all, I think it's kinda funny lol